src/v/coproc/reconciliation_backend.cc

```diff
@@ -225,10 +353,14 @@ reconciliation_backend::create_non_replicable_partition(
     auto copro_partition = _coproc_pm.local().get(ntp);
     if (likely(!copro_partition)) {
         // get_source_topic will assert if incorrect API is used
-        auto src_partition = _cluster_pm.local().get(model::ntp(
-          ntp.ns, tt_md->second.get_source_topic(), ntp.tp.partition));
+        model::ntp src_ntp(
+          ntp.ns, tt_md->second.get_source_topic(), ntp.tp.partition);
+        auto src_partition = _cluster_pm.local().get(src_ntp);
         if (!src_partition) {
-            co_return errc::partition_not_exists;
+            co_return stale_create_non_replicable_partition_request(
```

Comment:
can `stale_create_non_replicable_partition_request` return an error code ?

Reply:
No its either a stale request or not, so boolean